src/cargo/ops/cargo_new.rs

```diff
@@ -866,14 +867,40 @@ mod tests {
             }
         }
     }
-
-    if let Err(e) = Workspace::new(&path.join("Cargo.toml"), config) {
-        crate::display_warning_with_error(
-            "compiling this new package may not work due to invalid \
+    match Workspace::new(&path.join("Cargo.toml"), config) {
+        Ok(ws) => {
+            let root_manifest = paths::read(ws.root_manifest())?;
+            let workspace_document =
+                parse_document(root_manifest.as_str(), ws.root_manifest(), config)?;
+            match workspace_document["workspace"]["package"].as_table() {
+                Some(workspace_package_keys) => {
+                    if workspace_package_keys.len() == 0 {
+                        return Ok(());
+                    }
+                    let package_manifest = paths::read(&path.join("Cargo.toml"))?;
+                    let mut package_document =
+                        parse_document(package_manifest.as_str(), ws.root_manifest(), config)?;
+                    let package_table = package_document["package"].as_table_mut().unwrap();
+                    workspace_package_keys.iter().for_each(|(key, _)| {
```

Comment:
Still need to format the document and handle the `edition` field.

Reply:
When writing tests, [ui tests](https://doc.crates.io/contrib/tests/writing.html#ui-tests) would be a good candidate so we precisely know how the files will be formatted